com.unity.shadergraph/Editor/Data/Nodes/Input/PropertyNode.cs

```diff
@@ -151,7 +151,14 @@ public void GenerateNodeCode(ShaderStringBuilder sb, GenerationMode generationMo
                     sb.AppendLine($"$precision4 {GetVariableNameForSlot(OutputSlotId)} = {property.referenceName};");
                     break;
                 case PropertyType.Color:
-                    sb.AppendLine($"$precision4 {GetVariableNameForSlot(OutputSlotId)} = {property.referenceName};");
+                    if (property.version == 0)
```

Comment:
I think a switch statement here with each version explicitly spelled out would be a more future-proof way to do it. Imagine someone loading a graph a year from now that includes a V2 version - that should error in this version.

Reply:
agreed